app/bundles/AssetBundle/Controller/PublicController.php

```
@@ -1,92 +1,137 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Mautic\AssetBundle\Controller;
 
-use Mautic\CoreBundle\Controller\FormController as CommonFormController;
+use Doctrine\ORM\ORMException;
+use Mautic\AssetBundle\Entity\Asset;
+use Mautic\AssetBundle\Model\AssetModel;
+use Mautic\CoreBundle\Controller\AbstractFormController;
 use Mautic\CoreBundle\Helper\CoreParametersHelper;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
-class PublicController extends CommonFormController
+class PublicController extends AbstractFormController
 {
     /**
-     * @param string $slug
+     * Handles public download of assets by slug.
+     *
+     * This method performs the initial validation of the slug, retrieves the
+     *  corresponding Asset entity, and then delegates further response logic
+     *  (such as redirects, streaming, or access denial) to `createAssetResponse()`.
+     *
+     * @throws ORMException
+     */
+    public function downloadAction(
+        Request $request,
+        CoreParametersHelper $parametersHelper,
+        AssetModel $model,
+        string $slug,
+    ): Response {
+        try {
+            $entity = $model->getRepository()->findByIdAndAlias($slug);
+        } catch (\Exception) {
```

This isn't the use case to catch all exceptions. Please catch only the exception you want to catch. I doubt it's all of them. Code like this is hiding real issues from the logs.